packages/cli/lib/services/dryrun.service.ts

```diff
@@ -97,61 +97,72 @@ export class DryRunService {
-                            return integration.postConnectionScripts.some((postConnectionScript) => postConnectionScript === syncName);
-                        } else {
-                            return false;
-                        }
-                    })?.providerConfigKey || '';
-                isPostConnectionScript = true;
+        // Find the appropriate script to run
+        let syncInfo: ParsedNangoSync | ParsedNangoAction | undefined;
+        for (const integration of response.parsed.integrations) {
+            if (options.optionalProviderConfigKey && integration.providerConfigKey !== options.optionalProviderConfigKey) {
+                continue;
             }
 
-            if (!providerConfigKey) {
-                console.log(
-                    chalk.red(
-                        `Provider config key not found, please check that the provider exists for this sync name: ${syncName} by going to the Nango dashboard. (and pass it in as --integration-id)`
-                    )
-                );
-                return;
+            // Priority for syncs and actions
+            for (const script of [...integration.syncs, ...integration.actions]) {
+                if (script.name !== syncName) {
+                    continue;
+                }
+                if (syncInfo) {
+                    console.log(chalk.red(`Multiple integrations contain a script named "${syncName}". Please use "--integration-id"`));
+                    return;
+                }
+                syncInfo = script;
+                providerConfigKey = integration.providerConfigKey;
             }
-        }
-
-        const foundConfig = response.parsed.integrations.find((integration) => {
-            const syncsArray = integration.syncs || [];
-            const actionsArray = integration.actions || [];
 
-            return [...syncsArray, ...actionsArray].some((sync) => sync.name === syncName);
-        });
+            // If nothing that could still be a post connection script
+            if (!syncInfo) {
+                for (const script of integration.postConnectionScripts) {
+                    if (script !== syncName) {
+                        continue;
+                    }
+                    if (isPostConnectionScript) {
+                        console.log(chalk.red(`Multiple integrations contain a post connection script named "${syncName}". Please use "--integration-id"`));
+                        return;
+                    }
+                    isPostConnectionScript = true;
+                    providerConfigKey = integration.providerConfigKey;
+                }
+            }
+        }
 
-        const syncInfo = foundConfig
-            ? (foundConfig.syncs || []).find((sync) => sync.name === syncName) || (foundConfig.actions || []).find((action) => action.name === syncName)
-            : null;
+        if ((!syncInfo && !isPostConnectionScript) || !providerConfigKey) {
```

Comment:
can we not simply check for `!providerConfigKey`?

Reply:
it's ts pleasing 😬